src/client/Client.cc

```diff
@@ -2944,6 +2944,10 @@ void Client::send_reconnect(MetaSession *session)
 	cap.gen = session->cap_gen;
 	cap.issued = cap.implemented = CEPH_CAP_PIN;
       } else {
+	if (in->is_dir()) {
+	  // remove caps for async dir operations
+	  cap.implemented &= (CEPH_CAP_ANY_SHARED | CEPH_CAP_ANY_EXCL);
```

Comment:
Shouldn't this also include the rest of the FILE caps? What about DIR_CREATE and DIR_UNLINK?

Reply:
it's unlikely this will change in the future. because mds lose lock cache after restart